### PR TITLE
docs: add clickhouse blog

### PR DESCRIPTION
**Pull request by yzeng25**
Changes:

Add ClickHouse blog

### CODE REVIEW

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+> 本文讲述了社区贡献者齐振东为 Apache APISIX 贡献 `clickhouse-logger` 的历程，以及如何使用该插件简化业务架构，提升全链路日志效率。
+
+<!--truncate-->
+
+## 背景信息
+
+Apache APISIX 是一个动态、实时、高性能的 API 网关，提供负载均衡、动态上游、灰度发布、服务熔断、身份认证、可观测性等丰富的流量管理功能。作为 API 网关，Apache APISIX 拥有多种类型的实用插件。得益于 Apache APISIX 的插件热加载机制，我们无需重启便可启用插件、修改插件配置和停用插件。
+
+ClickHouse 由 Yandex 开发，在2016年开源。ClickHouse 不止是一个数据库， 也是一个数据库管理系统，它允许在运行时创建表和数据库、加载数据和运行查询，而无需重新配置或重启服务。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md]:
```suggestion
ClickHouse 由 Yandex 开发，在 2016 年开源。ClickHouse 不止是一个数据库， 也是一个数据库管理系统，它允许在运行时创建表和数据库、加载数据和运行查询，而无需重新配置或重启服务。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+          "http_user_agent": "$http_user_agent",
+          "request_time": "$request_time",
+          "upstream_addr": "$upstream_addr",
+          "http_host": "$http_host",
+          "content_type": "$content_type"
+      }
+  }'
+  ```
+
+以下是使用 Grafana  与 Clickhouse 对接后的仪表盘视图。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md]:
```suggestion
以下是使用 Grafana 与 Clickhouse 对接后的仪表盘视图。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+1. ClickHouse 提供 HTTP 接口，方便其它模块调用。
+2. 基于 ClickHouse 的分析工具链很成熟，能够满足我们对日志分析的需求。
+3. ClickHouse 支持使用对象存储作为存储引擎，非常方便。
+4. 没有必要自己重复“造轮子”。
+
+接下来就只剩下一个问题需要解决了：如何实现 Apache APISIX 和 ClickHouse 之间的对接？以插件的形式实现对接其实是一个不错的方法。作为 Apache APISIX 社区的一员，我一直都在社区里面“潜水”，看到了最近 Apache APISIX 在生态方面的持续进步，其实我也有些心动，一直在使用 Apache APISIX，但还没有给社区贡献过代码，不如就借这个机会为社区的生态发展添一把火吧。
+
+## ClickHouse 插件实现原理
+
+`clickhouse-logger` 插件的作为一个中间层，对接 Apache APISIX 和 ClickHouse。如前文所说，我们使用Apache APISIX 作为七层负载均衡，请求经过 Apache APISIX会产生日志，比如 access log 和 error log。`clickhouse-logger` 收集到日志后，会按照自身 metadata 所设置的日志格式，对这些日志进行整理。最后依靠批处理器将整理过的日志批量发送至 ClickHouse。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md]:
```suggestion
`clickhouse-logger` 插件的作为一个中间层，对接 Apache APISIX 和 ClickHouse。如前文所说，我们使用 Apache APISIX 作为七层负载均衡，请求经过 Apache APISIX会产生日志，比如 access log 和 error log。`clickhouse-logger` 收集到日志后，会按照自身 metadata 所设置的日志格式，对这些日志进行整理。最后依靠批处理器将整理过的日志批量发送至 ClickHouse。
```

File: website/blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        "nodes": {
+            "127.0.0.1:1980": 1
+        }
+    }
+}'
+```
+
+## Summary
+
+The above is the whole process of developing clickhouse-logger for Apache APISIX, and how to implement it with Grafana to achieve the same observability with simpler architechure and worlflows. I hope more people in the community will be willing to step out of their comfort zones. Switching from followers to contributors is much easier than you think.
```
Comment (SylviaBABY) [inline website/blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md]:
```suggestion
The above is the whole process of developing `clickhouse-logger` for Apache APISIX, and how to implement it with Grafana to achieve the same observability with simpler architechure and worlflows. I hope more people in the community will be willing to step out of their comfort zones. Switching from followers to contributors is much easier than you think.
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+
+随着越来越多的公司开始将业务迁移上云，如何高效地实现日志收发及日志分析，增强系统的可观测性成为了一个难题。中国移动作为一家提供公有云服务的企业，前期基于 Apache APISIX 的业务日志收发和和分析系统的架构大致是这样的。
+
+![initial bussiness architecture](https://static.apiseven.com/202108/1646363723740-f92d6a39-64e0-4464-8c44-c73832362bf6.png)
+
+随着业务的增长，上述方案不仅维护成本奇高，而且难以满足我们对于精细化数据分析的需求。由于 rsyslog 接收到的数据是一个字符串，而不是 JSON 格式的日志，给日志分析带来了一定的难度。
+
+计算机领域有句名言： “任何问题都可以通过增加一个间接的中间层来解决”。我们其实也考虑过在 `tcp-logger` 和 rsyslog 之间再加一个中间层，使字符串转换为 JSON。但这显然不是长久之计。
+
+所以我们换了一种思路去看待这个问题：如果把现有架构中的 “tcp-logger+rsyslog+Promtail+Loki” 看作是一个巨大的中间层，那么不论我们在这中间怎么添加额外的中间层，除了能解决燃眉之急外，只会使它变得更加臃肿和难以维护。市面上有没有一个产品能直接把“tcp-logger+rsyslog+Promtail+Loki”给替换掉呢？
```
Comment (guitu168) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md]:
```suggestion
所以我们换了一种思路去看待这个问题：如果把现有架构中的“tcp-logger+rsyslog+Promtail+Loki”看作是一个巨大的中间层，那么不论我们在这中间怎么添加额外的中间层，除了能解决燃眉之急外，只会使它变得更加臃肿和难以维护。市面上有没有一个产品能直接把“tcp-logger+rsyslog+Promtail+Loki”给替换掉呢？
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+
+接下来就只剩下一个问题需要解决了：如何实现 Apache APISIX 和 ClickHouse 之间的对接？以插件的形式实现对接其实是一个不错的方法。作为 Apache APISIX 社区的一员，我一直都在社区里面“潜水”，看到了最近 Apache APISIX 在生态方面的持续进步，其实我也有些心动，一直在使用 Apache APISIX，但还没有给社区贡献过代码，不如就借这个机会为社区的生态发展添一把火吧。
+
+## ClickHouse 插件实现原理
+
+`clickhouse-logger` 插件的作为一个中间层，对接 Apache APISIX 和 ClickHouse。如前文所说，我们使用Apache APISIX 作为七层负载均衡，请求经过 Apache APISIX 会产生日志，比如 access log 和 error log。`clickhouse-logger` 收集到日志后，会按照自身 metadata 所设置的日志格式，对这些日志进行整理。最后依靠批处理器将整理过的日志批量发送至 ClickHouse。
+
+![clickhouse-logger architecture](https://static.apiseven.com/202108/1646363936994-c2646095-1ea4-4c1f-8cad-1dcecfc41df3.png)
+
+`clickhouse-logger` 在我们这个场景下，起到了替代 “tcp-logger+rsyslog+Promtail+Loki”的作用。免除了多个组件之间的格式转换和数据转发，可将 Log 数据请求直接推送到 ClickHouse 服务器。
```
Comment (guitu168) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md]:
```suggestion
`clickhouse-logger` 在我们这个场景下，起到了替代“tcp-logger+rsyslog+Promtail+Loki”的作用。免除了多个组件之间的格式转换和数据转发，可将 Log 数据请求直接推送到 ClickHouse 服务器。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+        "nodes": {
+            "127.0.0.1:1980": 1
+        }
+    }
+}'
+```
+
+## 总结
+
+以上就是我为 Apache APISIX 开发 `clickhouse-logger` 的全过程。希望社区里有更多的人愿意走出舒适区，实现自身的角色转换，从关注者变为贡献者的远比你想象的简单。
```
Comment (guitu168) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md]:
```suggestion
以上就是我为 Apache APISIX 开发 `clickhouse-logger` 的全过程。希望社区里有更多的人愿意走出舒适区，实现自身的角色转换，从关注者变为贡献者的过程远比你想象的简单。
```

**Review by guitu168 — DISMISSED** [review]

**Review by SylviaBABY — DISMISSED** [review]

**Review by oil-oil — DISMISSED** [review]

File: website/blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md [hunk 8]
```diff
@@ -0,0 +1,248 @@
+---
+title: "Making It Simple, Apache APISIX Integrates ClickHouse to Improve Logging Efficiency"
```
Comment (juzhiyuan) [inline website/blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md]: How about using `api gateway` instead of `apisix` in url?

Comment (yzeng25) [reply website/blog/2022/03/04/apisix-clickhouse-makes-logging-easier.md]: updated both file names

**Review by SylviaBABY — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**